### PR TITLE
fix: zsh completion to work with source

### PR DESCRIPTION
**Pull request by btoonk**

### CODE REVIEW

**Review by waltzofpearls — APPROVED** [review]
Great work and approved!

I left 2 comments about variable naming. One letter variable vs a more readable name. This is debatable, since I also use one letter variables for method receiver variables.

**Comment by waltzofpearls** [comment]
Anther thing to mention, I tried to tested it since I have bash, and I haven't been able to get completion working for flags. I tried upgrading from default bash (v3) to v5, and upgraded bash-completion from v1 to v2. Also tried hardcode completion results. Trying to figure out what I'm missing in my bash config.

**Comment by btoonk** [comment]
@waltzofpearls The newer bash version is indeed required. You also need to install bash completion (brew install bash-completion@2) and after that the following in your .bashrc should do the trick:

```
  [[ -r "/opt/homebrew/etc/profile.d/bash_completion.sh" ]] && . "/opt/homebrew/etc/profile.d/bash_completion.sh"

source <(mysocketctl completion bash)
```

File: cmd/tunnel.go [hunk 1]
```diff
@@ -154,6 +155,33 @@ var tunnelConnectCmd = &cobra.Command{
 	},
 }
 
+func getTunnels(toComplete string) []string {
+	var r []string
```
[the remark this answers is not in the document]
Comment (btoonk) [reply cmd/tunnel.go]: changed the variable to tunnelIDs

File: cmd/socket.go [hunk 2]
```diff
@@ -223,6 +223,29 @@ var socketShowCmd = &cobra.Command{
 	},
 }
 
+func getSockets(toComplete string) []string {
+	var r []string
```
[the remark this answers is not in the document]
Comment (btoonk) [reply cmd/socket.go]: changed the variable to socketIDs.

**Pull request merged**